### ISSUE TITLE
example app crashes on disconnect (ios only)

### ISSUE THREAD

**Issue by marcmo** [Polidea/FlutterBleLib#25]
steps to reproduce:
build included example, run on ios, connect -> disconnect

seems you get your types wrong when calling `convertToBleDeviceMessage`. this is the function:
```
+ (BleDataServiceMessage* _Nonnull) convertBleDataServiceMessage: (id _Nonnull) value {
    BleDataServiceMessage* bleDataServiceMessage = [[BleDataServiceMessage alloc] init];
    bleDataServiceMessage.id_p = ((NSNumber *) value[@"id"]).doubleValue;
    bleDataServiceMessage.device =  [[BleDataBleDeviceMessage alloc] init];
    bleDataServiceMessage.device.id_p = value[@"deviceID"];
    bleDataServiceMessage.uuid = value[@"uuid"];
    bleDataServiceMessage.isPrimary = value[@"isPrimary"];
    
    return bleDataServiceMessage;
}
```
connecting you handle it correclty:
![image](https://user-images.githubusercontent.com/99541/41588481-2d361324-73b2-11e8-84a1-d3a30fd6b57a.png)
but not so much when disconnecting:
![image](https://user-images.githubusercontent.com/99541/41588521-4fe4055c-73b2-11e8-8010-a80bb6333566.png)

I saw you have started some major refactorings for ios on the `refactoring_ios` branch...are you still thinking about continuing your efforts? seems this has gone stale

**Comment by NOA-Charles** [comment]
Is this problem solved? I also encountered the same problem.

**Comment by NOA-Charles** [comment]
Dude, I saw your update in pull requests.

**Comment by NOA-Charles** [comment]
This solves the problem, thank you

**Comment by pawelByszewski** [comment]
@marcmo It has been a long time since we stopped maintaining the library but now we come with great news! We are currently working with Frontside on a new BLE Flutter library and BLE simulator. Beta version has been released already and we are going to release 2.0.0 in the first half of November
